src/glue/jobs/s3_to_json.py

```diff
@@ -0,0 +1,209 @@
+                May be None if filename only contains the `end_date`.
+            end_date (datetime.datetime): The date of the most recent data collected.
+            subtype (str): If this is HealthKitV2Samples, the type of the
+                sample data. (Not to be confused with the data type, which in
+                this case is HealthKitV2Samples or HealthKitV2Samples_Deleted)
+    """
+    logger = logging.getLogger(__name__)
+    metadata = {}
+    basename_components = os.path.splitext(basename)[0].split("_")
+    metadata["type"] = basename_components[0]
+    if "-" in basename_components[-1]:
+        start_date, end_date = basename_components[-1].split("-")
+        metadata["start_date"] = \
+                datetime.datetime.strptime(start_date, "%Y%m%d")
+        metadata["end_date"] = \
+                datetime.datetime.strptime(end_date, "%Y%m%d")
+    else:
+        metadata["start_date"] = None
+        metadata["end_date"] = \
+                datetime.datetime.strptime(basename_components[-1], "%Y%m%d")
+    if metadata["type"] == "HealthKitV2Samples":
+        metadata["subtype"] = basename_components[1]
+    if (
+            metadata["type"] == "HealthKitV2Samples"
+            and basename_components[-2] == "Deleted"
+        ):
+        metadata["type"] = "HealthKitV2Samples_Deleted"
+    logger.debug(f"metadata = {metadata}")
+    return metadata
+
+def process_record(s3_obj, workflow_run_properties):
+    """
+    Write the contents of a .zip archive stored on S3 to their respective
+    JSON dataset.
+
+    Metadata derived from the filename is inserted into each JSON before
+    being written to its JSON dataset.
+
+    Args:
+        s3_obj (dict): An S3 object as returned by `boto3.get_object`.
+        workflow_run_properties (dict): The workflow arguments
+
+    Returns:
+        None
+    """
+    logger = logging.getLogger(__name__)
+    with zipfile.ZipFile(io.BytesIO(s3_obj["Body"])) as z:
+        non_empty_contents = [
+                f.filename for f in z.filelist
+                if "/" not in f.filename and
+                "Manifest" not in f.filename and
+                f.file_size > 0
+        ]
+        logger.debug(f'contents: {z.namelist()}')
+        logger.debug(f'non-empty contents: {non_empty_contents}')
+        for json_path in non_empty_contents:
+            metadata = get_metadata(os.path.basename(json_path))
+            dataset_identifier = metadata["type"]
+            logger.info(f"Writing {json_path} to dataset {dataset_identifier}")
+            write_file_to_json_dataset(
+                    z=z,
+                    json_path=json_path,
+                    dataset_identifier=dataset_identifier,
+                    metadata=metadata,
+                    workflow_run_properties=workflow_run_properties)
+
+def main():
+    # Configure logger
+    logging.basicConfig()
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.DEBUG)
+
```

Comment:
Just out of curiosity - Is there a reason why you set the logger here and now on top of the module?

Reply:
This logger is scoped to the `main` function. So each function needs to get the existing logger (which is only configured once here) if they want to use logger methods.